source/guides/azure/infiniband.md

```diff
@@ -0,0 +1,488 @@
+GPU2    NV1     NV2     X       NV1     NV2     NODE    NODE    NODE    0-19            0
+GPU3    NV2     NV1     NV1     X       NODE    NV2     NODE    NODE    0-19            0
+GPU4    NODE    NODE    NV2     NODE    X       NV1     NV1     NV2     0-19            0
+GPU5    NODE    NODE    NODE    NV2     NV1     X       NV2     NV1     0-19            0
+GPU6    NV1     NODE    NODE    NODE    NV1     NV2     X       NV2     0-19            0
+GPU7    NODE    NV1     NODE    NODE    NV2     NV1     NV2     X       0-19            0
+
+Legend:
+
+  X    = Self
+  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
+  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
+  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
+  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
+  PIX  = Connection traversing at most a single PCIe bridge
+  NV#  = Connection traversing a bonded set of # NVLinks
+
+```
+
+## Install UCX-Py and tools
+
+```shell
+wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+bash Miniconda3-latest-Linux-x86_64.sh
+```
+
+Accept the default and allow conda init to run. Then start a new shell.
+
+Create a conda environment (see ucx-py docs)
+
+```shell
+conda create -n ucxpy -c conda-forge -c rapidsai python=3.7 ipython ucx-proc=*=gpu ucx ucx-py dask distributed numpy cupy pytest pynvml rmm cudf dask-cudf dask-cuda -y
+...
+conda activate ucxpy
+```
+
+Clone ucx-py repo locally
+
+```shell
+git clone https://github.com/rapidsai/ucx-py.git
+cd ucx-py/benchmarks
+```
+
+## Run Tests
+
+Start by running the UCX-Py test suite, from within the `ucx-py` repo:
+
+```shell
+pytest -vs tests/
+pytest -vs ucp/_libs/tests/
+```
+
+Now check to see if IB works, for that you can run some of the benchmarks that we include in UCX-Py, for example:
+
+```shell
+# Let UCX pick the best transport (expecting NVLink when available,
+# otherwise IB, or TCP in worst case) on devices 0 and 1
+python -m ucp.benchmarks.send_recv --server-dev 0 --client-dev 1 -o rmm --reuse-alloc -n 128MiB
+
+# Force TCP-only on devices 0 and 1
+UCX_TLS=tcp,cuda_copy python -m ucp.benchmarks.send_recv --server-dev 0 --client-dev 1 -o rmm --reuse-alloc -n 128MiB
+```
+
+We expect the first case above to have much higher bandwidth than the second. If you happen to have both
+NVLink and IB connectivity, then you may limit to the specific transport by specifying `UCX_TLS`, e.g.:
+
+```shell
+# NVLink (if available) or TCP
+UCX_TLS=tcp,cuda_copy,cuda_ipc
+
+# IB (if available) or TCP
+UCX_TLS=tcp,cuda_copy,rc
+
+```
+
+## Run Benchmarks
+
+Finally, let's run the [merge benchmark](https://github.com/rapidsai/dask-cuda/blob/branch-22.12/dask_cuda/benchmarks/local_cudf_merge.py) from `dask-cuda`.
```

Comment:
@pentschev @rjzamora Do we have anything written up on why communication is important for this benchmark?

Reply:
Good question. A merge operation is essentially two shuffle operations (one for each "side" of the merge), followed by a partition-wise merge. Therefore, it is the "all-to-all" shuffle operation that makes communication important. Especially at scale.

Your question is whether this is clearly explained in any documentation: I am honestly not sure if it is (I am not personally aware of anything like this in the docs).

Reply:
Okay. I will attempt to synthesize an explanation here. Using what you've described and what we have in the dask docs [here](https://docs.dask.org/en/stable/dataframe-groupby.html#) and [here](https://docs.dask.org/en/stable/dataframe-best-practices.html#avoid-full-data-shuffling).